File: app/src/test/java/k/bs/designpatternsp/behavoral/composite/ja/CompositeTest.java
package k.bs.designpatternsp.behavoral.composite.ja;

import org.hamcrest.CoreMatchers;
import org.junit.Assert;
import org.junit.Test;

import k.bs.designpatternsp.structural.composite.ja.computer.HardDrive;
import k.bs.designpatternsp.structural.composite.ja.computer.Memory;
import k.bs.designpatternsp.structural.composite.ja.computer.PersonalComputer;
import k.bs.designpatternsp.structural.composite.ja.computer.Processor;

public class CompositeTest {

    @Test
    public void compositionTest() {
        PersonalComputer pc = new PersonalComputer("PC");

        pc.add(new Processor(1000, "Intel"));
        pc.add(new Memory(500, "8GB"));
        pc.add(new HardDrive(800, "SSD"));

        Assert.assertThat(pc.getName(), CoreMatchers.is("PC"));
        Assert.assertThat(pc.getPrice(), CoreMatchers.is(2300));
    }
}

File: app/src/test/java/k/bs/designpatternsp/behavoral/composite/kt/CompositeTest.kt
package k.bs.designpatternsp.behavoral.composite.kt

import k.bs.designpatternsp.structural.composite.kt.computer.HardDrive
import k.bs.designpatternsp.structural.composite.kt.computer.Memory
import k.bs.designpatternsp.structural.composite.kt.computer.PersonalComputer
import k.bs.designpatternsp.structural.composite.kt.computer.Processor
import org.hamcrest.CoreMatchers
import org.junit.Assert
import org.junit.Test

class CompositeTest {

    @Test
    fun `Composite`() {
        val pc = PersonalComputer()
            .add(Processor())
            .add(HardDrive())
            .add(Memory())

        println(pc.price)

        Assert.assertThat(pc.name, CoreMatchers.equalTo("PC"))
        Assert.assertThat(pc.price, CoreMatchers.equalTo(1600))
    }
}